### PR TITLE
[Distributed] Handle distributed thunk requirements/witnesses

### PR DESCRIPTION
**Pull request by ktoso**
This is an important piece for allowing calls on `@DistributedProtocol` protocols -- and marks our first "complete" roundtrip call through a protocol!

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
Reviews welcome, but merging so we don't have so many in-flight-overlapping PRs. I'd address any feedback in follow-up 🙏

**Comment by ktoso** [comment]
unrelated failure 
```
******************** TEST 'lldb-api :: lang/swift/embedded/expr/TestSwiftEmbeddedExpression.py' FAILED ********************

```

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS

File: lib/IRGen/GenDistributed.cpp [hunk 1]
```diff
@@ -858,6 +858,9 @@ FunctionPointer AccessorTarget::getPointerToTarget(llvm::Value *actorSelf) {
   }
 
   auto *requirementDecl = Target.get<AbstractFunctionDecl *>();
+  if (requirementDecl->isDistributed()) {
+    requirementDecl = requirementDecl->getDistributedThunk();
```
Comment (xedin) [inline lib/IRGen/GenDistributed.cpp:862]: I don't think this is right, accessor should be already anchored on the "thunk" of the requirement, we need to figure out why it's not and fix it in GenMeta or IRGenSIL

Comment (ktoso) [reply lib/IRGen/GenDistributed.cpp:862]: Good news -- this was dead code, removed.

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Pull request merged**